app/controllers/api/v2/base_controller.rb

```diff
@@ -155,6 +157,13 @@ def render_error(error, options = { })
                              :layout   => 'api/v2/layouts/error_layout')
       end
 
+      def show_errors(exception = nil)
+        if exception
+          e = exception.message.scan(/\(([^)]+)\)/)
+          not_found _("Associated resource %{resource} with id %{id} not found") % {:resource => e[0][0], :id => e[1][0]}
```

Comment:
This generates a message `Associated resource subnet_id with id 10000 not found`, note the resource is subnet_id not subnet. Either we should change the message to `Associated resource could not be found by key subnet_id with id 10000` or change the code so we print the resource, e.g. Subnet. In fact I think both resource name and the used key would be interesting.